Firebase.h

```diff
@@ -25,49 +25,97 @@
 #include <WiFiClientSecure.h>
 #include <ESP8266HTTPClient.h>
 
-// FirebaseError represents a Firebase API error with a code and a
-// message.
-class FirebaseError {
- public:
-  operator bool() const { return _code < 0; }
-  int code() const { return _code; }
-  const String& message() const { return _message; }
-  void reset() { set(0, ""); }
-  void set(int code, const String& message) {
-    _code = code;
-    _message = message;
-  }
- private:
-  int _code = 0;
-  String _message = "";
-};
+//TODO(edcoyne) split these into multiple files.
+
+class FirebaseCall;
+class FirebaseEventStream;
 
-// Firebase is the connection to firebase.
+// Primary client to the Firebase backend.
 class Firebase {
  public:
   Firebase(const String& host);
   Firebase& auth(const String& auth);
-  const FirebaseError& error() const {
-    return _error;
+
+  // Fetch result at "path" to a local variable. If the value is too large you will exceed
+  // local memory.
+  FirebaseCall get(const String& path);
```

Comment:
The problem with not handing the Call object back to the client is the lifetime of the call object is bound now to the lifetime of the Firebase object. One should be short lived and one long lived.

For example, if you call Firebase.get() then never read the response we can never clear the response from the buffer because we don't know if you will ask for it 3 hours later. However if the client owns the FirebaseCall object and deletes it, we can then clear the response from the buffer because it will never be read. 

Let me ponder on it for a bit and see if there is a way we can get the ease of use that you want with the lifetime isolation I want.